modules/sentry/services/sentry.nix

```diff
@@ -0,0 +1,785 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.services.sentry;
+
+  uwsgiWithPython =
+    (pkgs.uwsgi.override { plugins = [ "python2" ]; }).overrideAttrs (oldDrv: {
+      buildInputs = oldDrv.buildInputs ++ [ pkgs.makeWrapper ];
+
+      installPhase = oldDrv.installPhase + ''
+        runHook postInstall
+      '';
+
+      postInstall = (oldDrv.postInstall or "") + ''
+        mv $out/bin/uwsgi $out/bin/uwsgi-wrapped
+        makeWrapper "$out/bin/uwsgi-wrapped" "$out/bin/uwsgi" \
+          --add-flags "--plugin $out/lib/uwsgi/python2_plugin.so"
+      '';
+    });
+
+  surround = prefix: suffix: x: if x != null then prefix + x + suffix else null;
+
+  or_ = a: b: if a != null then a else (if b != null then b else null);
+
+in {
+  options.services.sentry = {
+    enable = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Whether to run the sentry suite of services.
+      '';
+    };
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.sentry;
+      example = literalExample "pkgs.sentry";
+      description = ''
+        The sentry package to use.
+      '';
+    };
+
+    hostname = mkOption {
+      type = types.str;
+      default = "0.0.0.0";
+      description = ''
+        Hostname the Sentry web server should bind to.
+      '';
+    };
+
+    eventRetentionDays = mkOption {
+      type = types.int;
+      default = 90;
+      description = ''
+        Sentry has a cleanup cron job the prunes events older than this many days.
+      '';
+    };
+
+    dbName = mkOption {
+      type = types.str;
+      default = "sentrydb";
+      description = ''
+        Name of the Sentry database.
+      '';
+    };
+
+    dbUser = mkOption {
+      type = types.str;
+      default = "sentry";
+      description = ''
+        Sentry database user name.
+      '';
+    };
+
+    dbPassword = mkOption {
+      type = with types; nullOr str;
+      default = null;
+      description = ''
+        Password for Sentrys postgresql database.
+        This setting will be ignored if dbPasswordFile is set.
+        Using this option will store the password in plain text in the
+        world-readable nix store. To avoid this the <literal>dbPasswordFile</literal> can be used.
+      '';
+    };
+
+    dbPasswordFile = mkOption {
+      type = with types; nullOr path;
+      default = null;
+      description = ''
+        Password file for the Sentrys postgresql database.
+      '';
+    };
+
+    port = mkOption {
+      type = types.int;
+      default = 9999;
+      description = ''
+        Port the Sentry web server should bind to.
+      '';
+    };
+
+    secretKey = mkOption {
```

Comment:
I would completely remove this option, and only allow `secretKeyFile`.

Reply:
Fixed.